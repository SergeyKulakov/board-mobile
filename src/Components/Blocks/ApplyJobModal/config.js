import { getTranslate } from 'Helpers/languages'

export const getText = () => ({
  title: getTranslate('jobDetail.applyForThisJob'),
  subTitle: getTranslate('jobDetail.suggestWhichDay'),
  apply: getTranslate('findJobPage.apply'),
  skip: getTranslate('jobPost.cancel'),
})
