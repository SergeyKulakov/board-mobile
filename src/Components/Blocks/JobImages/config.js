export default {
  defaultImageProps: {
    resizeMode: 'cover',
    withLoading: true,
  },
}
