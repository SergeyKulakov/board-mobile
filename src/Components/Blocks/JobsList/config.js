import { Platform } from 'react-native'

export default {
  SwipeListView: {
    leftOpenValue: 150,
    swipeToOpenPercent: 70,
    rightOpenValue: -160,
    closeOnRowOpen: true,
    stopLeftSwipe: 160,
    stopRightSwipe: -170,
    useFlatList: true,
    closeOnRowPress: true,
    // FlatList props
    recalculateHiddenLayout: true,
    onEndReachedThreshold: Platform.select({ ios: 0.4, android: 0.5 }),
    keyExtractor: item => item._id,
  },
}
