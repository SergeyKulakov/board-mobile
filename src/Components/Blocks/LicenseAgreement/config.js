import { termsInfo } from 'Config/amplify'
import { appBundleId, clientUrl } from 'Constants/api'

export const requestUrl = `${clientUrl}/terms-conditions?redirect_uri=${appBundleId}://oauth`
