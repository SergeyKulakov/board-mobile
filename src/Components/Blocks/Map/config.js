export default {
  mapType: 'standard',
  showsUserLocation: true,
  zoomEnabled: true,
  followUserLocation: true,
}
