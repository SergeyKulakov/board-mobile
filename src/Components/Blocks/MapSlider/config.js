import { Dimensions } from 'react-native'
import metrics from 'Themes/Metrics'

export default {
  SnapCarousel: {
    getSliderWidth: () => Dimensions.get('window').width,
    getItemWidth: () => metrics.widthPercentageToDP(100),
  },
}
