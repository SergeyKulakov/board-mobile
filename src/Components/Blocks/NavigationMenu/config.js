import * as routes from 'Constants/routes'

export const data = [
  {
    icon: 'search',
    text: 'headerBlock.menu.findJob',
    key: routes.findJobs,
  },
  {
    icon: 'pencil-square-o',
    text: 'headerBlock.menu.postJob',
    key: routes.postJob,
  },
  {
    icon: {
      type: 'mci',
      name: 'worker',
    },
    text: 'headerBlock.menu.findHelp',
    key: routes.findHelp,
  },
  {
    icon: 'list-alt',
    text: 'headerBlock.menu.myJobs',
    key: routes.myJobs,
  },
  {
    icon: 'heart',
    text: 'headerBlock.menu.myFavourites',
    key: routes.myFavourites,
  },
  {
    icon: 'bell',
    text: 'homePage.Notifications',
    key: routes.notifications,
  },
  {
    icon: 'comments',
    text: 'homePage.chats',
    key: routes.chats,
  },
  {
    icon: 'location-arrow',
    text: 'headerBlock.menu.myRequests',
    key: routes.myRequests,
  },
  {
    icon: {
      type: 'fa5',
      name: 'search-location',
    },
    text: 'headerBlock.menu.trackNow',
    key: routes.track,
  },
  {
    icon: 'user-circle',
    text: 'common.YourProfile',
    key: routes.profile,
  },
  {
    // image: wallet,
    icon: {
      name: 'wallet',
      type: 'ant',
    },
    text: 'common.subscription',
    key: routes.subscriptions,
  },
  {
    icon: 'cog',
    text: 'common.setting',
    key: routes.settings,
  },
]
