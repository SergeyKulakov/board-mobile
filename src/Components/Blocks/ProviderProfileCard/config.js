import i18n from 'I18N'

export const getText = () => ({
  jobsMore: i18n.t('serviceProvider.jobsDone'),
  hireNow: i18n.t('serviceProvider.hireNow'),
  more: i18n.t('serviceProvider.more'),
  getServiceTitle: value => i18n.t(`services.${value}`),
})
