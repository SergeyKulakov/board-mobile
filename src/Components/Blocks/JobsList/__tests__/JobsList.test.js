import React from 'react'
import { shallow } from 'enzyme'

import 'jest-styled-components'

import Component from '../JobsList'
import config from '../config'

describe('JobsList', () => {
  const defaultProps = {
    data: [],
    isPaginationLoading: false,
    isRefreshing: () => null,
    loadingFavoriteId: false,
    onApplyClick: () => null,
    onFavoriteClick: () => null,
    onLoadMoreVacancies: () => null,
    onRefresh: () => null,
    onRemoveVacancy: () => null,
    onShareVacancyClick: () => null,
    onVacancyClick: () => null,
    orientation: 'orientation',
    renderItem: undefined,
    user: {
      username: 'userId',
    },
    extraData: [],
    onShowAd: () => null,
    isAdsLoading: false,
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    const renderHiddenItems = wrapper.instance().renderHiddenItems({
      item: {
        _id: 'jobId',
        favouriteId: 'favouriteId',
        author: 'author',
      },
    })

    const renderHeaderComponent = wrapper.instance().renderHeaderComponent()

    const renderRow = wrapper.instance().renderRow(
      {
        item: {
          _id: 'jobId',
          author: 'userId',
        },
      },
      'rowMap',
    )

    const renderListAds = wrapper.instance().renderRow(
      {
        item: 'ads',
      },
      'rowMap',
    )

    expect([
      wrapper,
      renderHiddenItems,
      renderHeaderComponent,
      renderRow,
      renderListAds,
    ]).toMatchSnapshot()
  })

  it('close row', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    wrapper.instance().openedItemId = 1000

    wrapper
      .find('SwipeListView')
      .props()
      .onRowClose()

    expect(wrapper.instance().openedItemId).toBeNull()
  })

  it('remove card', done => {
    const props = {
      ...defaultProps,
      onRemoveVacancy: jest.fn(),
      data: [],
    }
    const getAnimate = () =>
      jest.fn(
        () =>
          new Promise(resolve => {
            setTimeout(resolve, 1000)
          }),
      )

    const wrapper = shallow(<Component {...props} />)

    const hiddenActionsRef = {
      '41er23r21f1': { animate: getAnimate() },
      '41er212312d': { animate: getAnimate() },
      '41ee1123sss': { animate: getAnimate() },
    }

    wrapper.instance().hiddenActions = hiddenActionsRef

    const jobsRef = {
      '41er23r21f1': { animate: getAnimate() },
      '41er212312d': { animate: getAnimate() },
      '41ee1123sss': { animate: getAnimate() },
    }

    wrapper.instance().jobs = jobsRef

    wrapper.instance().removedItemId = '41er23r21f1'

    const args = [
      '41er23r21f1',
      {
        '41er23r21f1': {
          closeRow: jest.fn(),
          props: { item: { _id: '41er23r21f1' } },
        },
        '41er212312d': {
          closeRow: jest.fn(),
          props: { item: { _id: '41er212312d' } },
        },
        '41ee1123sss': {
          closeRow: jest.fn(),
          props: { item: { _id: '41ee1123sss' } },
        },
      },
      config.SwipeListView.leftOpenValue,
    ]

    wrapper
      .find('SwipeListView')
      .props()
      .onRowOpen(...args)

    expect(args[1]['41er23r21f1'].closeRow).toHaveBeenCalledTimes(1)
    expect(args[1]['41er212312d'].closeRow).not.toHaveBeenCalled()
    expect(args[1]['41ee1123sss'].closeRow).not.toHaveBeenCalled()

    expect(hiddenActionsRef['41er23r21f1'].animate).toHaveBeenCalledTimes(1)
    expect(hiddenActionsRef['41er212312d'].animate).not.toHaveBeenCalled()
    expect(hiddenActionsRef['41ee1123sss'].animate).not.toHaveBeenCalled()

    expect(jobsRef['41er23r21f1'].animate).toHaveBeenCalledTimes(1)
    expect(jobsRef['41er212312d'].animate).not.toHaveBeenCalled()
    expect(jobsRef['41ee1123sss'].animate).not.toHaveBeenCalled()

    expect(props.onRemoveVacancy).not.toHaveBeenCalled()

    setTimeout(() => {
      expect(props.onRemoveVacancy).toHaveBeenCalledTimes(1)
      expect(props.onRemoveVacancy).toHaveBeenCalledWith({ _id: '41er23r21f1' })
      done()
    }, 1100)
  })

  it('set remove item id', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    expect(wrapper.instance().removedItemId).toBeNull()

    wrapper
      .find('SwipeListView')
      .props()
      .onSwipeValueChange({
        isOpen: true,
        key: '41er23r21f1',
        value: 150,
      })

    expect(wrapper.instance().removedItemId).toBe('41er23r21f1')

    wrapper
      .find('SwipeListView')
      .props()
      .onSwipeValueChange({
        isOpen: false,
        key: '41er23r21f1',
        value: 130,
      })

    expect(wrapper.instance().removedItemId).toBe('41er23r21f1')

    wrapper
      .find('SwipeListView')
      .props()
      .onSwipeValueChange({
        isOpen: false,
        key: '41er23r21f1',
        value: 160,
      })

    expect(wrapper.instance().removedItemId).toBeNull()
  })

  it('vacancy click action', () => {
    const props = {
      ...defaultProps,
      onVacancyClick: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    expect(props.onVacancyClick).not.toHaveBeenCalled()

    wrapper.instance().openedItemId = '2314ewqdsa'

    const args = [
      {
        item: {
          _id: '2314ewqdsa',
        },
      },
      {
        '2314ewqdsa': {
          closeRow: jest.fn(),
        },
      },
    ]

    wrapper.instance().handleVacancyClick(...args)

    expect(props.onVacancyClick).not.toHaveBeenCalled()
    expect(args[1]['2314ewqdsa'].closeRow).toHaveBeenCalledTimes(1)

    wrapper.instance().openedItemId = null

    wrapper.instance().handleVacancyClick(...args)

    expect(props.onVacancyClick).toHaveBeenCalledTimes(1)
    expect(args[1]['2314ewqdsa'].closeRow).toHaveBeenCalledTimes(1)
  })

  it('call load more action', () => {
    const props = {
      ...defaultProps,
      onLoadMoreVacancies: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    wrapper.instance().isScrollToTop = true

    expect(props.onLoadMoreVacancies).not.toHaveBeenCalled()

    const args = {
      distanceFromEnd: 0,
    }

    wrapper.instance().handleLoadMore(args)

    expect(props.onLoadMoreVacancies).not.toHaveBeenCalled()

    wrapper.instance().isScrollToTop = false

    wrapper.instance().handleLoadMore(args)

    expect(props.onLoadMoreVacancies).not.toHaveBeenCalled()

    const args2 = {
      distanceFromEnd: 1,
    }

    wrapper.instance().handleLoadMore(args2)

    expect(props.onLoadMoreVacancies).toHaveBeenCalledTimes(1)
  })
})
