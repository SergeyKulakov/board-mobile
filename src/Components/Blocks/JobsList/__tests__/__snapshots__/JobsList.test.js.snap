// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsList render snapshot 1`] = `
Array [
  <SwipeListView
    ListEmptyComponent={<UNDEFINED />}
    ListFooterComponent={
      <UNDEFINED
        visible={false}
      />
    }
    ListHeaderComponent={[Function]}
    closeOnRowBeginSwipe={false}
    closeOnRowOpen={true}
    closeOnRowPress={true}
    closeOnScroll={true}
    data={Array []}
    directionalDistanceChangeThreshold={2}
    disableLeftSwipe={false}
    disableRightSwipe={false}
    extraData={
      Array [
        "orientation",
        false,
        [Function],
        false,
      ]
    }
    keyExtractor={[Function]}
    leftOpenValue={150}
    onEndReached={[Function]}
    onEndReachedThreshold={0.4}
    onRefresh={[Function]}
    onRowClose={[Function]}
    onRowOpen={[Function]}
    onScroll={[Function]}
    onSwipeValueChange={[Function]}
    previewFirstRow={false}
    recalculateHiddenLayout={true}
    refreshing={[Function]}
    renderHiddenItem={[Function]}
    renderItem={[Function]}
    rightOpenValue={-160}
    stopLeftSwipe={160}
    stopRightSwipe={-170}
    swipeToClosePercent={50}
    swipeToOpenPercent={70}
    swipeToOpenVelocityContribution={0}
    useFlatList={true}
  />,
  <withAnimatable(View)
    animation="slideInRight"
    delay={0}
    direction="normal"
    iterationCount={1}
    iterationDelay={0}
    onAnimationBegin={[Function]}
    onAnimationEnd={[Function]}
    onTransitionBegin={[Function]}
    onTransitionEnd={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    useNativeDriver={false}
  >
    <ForwardRef>
      <UNDEFINED />
      <UNDEFINED
        isActiveFavorite={true}
        isFavouriteCallable={true}
        loading={false}
        onFavoriteClick={[Function]}
        onShareClick={[Function]}
      />
    </ForwardRef>
  </withAnimatable(View)>,
  <ForwardRef>
    <UNDEFINED
      isLoading={false}
      onShowAd={[Function]}
    />
  </ForwardRef>,
  <withAnimatable(View)
    animation="slideInRight"
    delay={0}
    direction="normal"
    iterationCount={1}
    iterationDelay={0}
    onAnimationBegin={[Function]}
    onAnimationEnd={[Function]}
    onTransitionBegin={[Function]}
    onTransitionEnd={[Function]}
    style={
      Object {
        "minHeight": 220,
      }
    }
    useNativeDriver={false}
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <ForwardRef>
        <UNDEFINED
          _id="jobId"
          author="userId"
          onShareClick={[Function]}
        />
      </ForwardRef>
    </TouchableWithoutFeedback>
  </withAnimatable(View)>,
  <UNDEFINED
    isLoading={false}
    onShowAd={[Function]}
    type="ads"
  />,
]
`;
